plutus-ledger-api/src/Plutus/V1/Ledger/EvaluationContext.hs

```diff
@@ -0,0 +1,46 @@
+module Plutus.V1.Ledger.EvaluationContext
+    ( EvaluationContext (..)
+    , mkEvaluationContext
+    , LedgerCostModel
+    , validateLedgerCostModel
+    ) where
+
+import PlutusCore as Plutus (DefaultFun, DefaultUni, defaultCekCostModel)
+import PlutusCore.Evaluation.Machine.BuiltinCostModel as Plutus
+import PlutusCore.Evaluation.Machine.CostModelInterface as Plutus
+import PlutusCore.Evaluation.Machine.MachineParameters as Plutus
+import UntypedPlutusCore.Evaluation.Machine.Cek as Plutus
+
+import Data.Map as Map
+import Data.Maybe
+import Data.Text qualified as Text
+
+{-| the (latest) raw/untyped protocol cost model
+We call it raw/untyped as it is just  dictionary of strings to cost integer values.
+
+synonym to ledger's `Alonzo.CostModel`
+synonym to `Plutus.CostModelParams`
+-}
+type LedgerCostModel = Map.Map Text.Text Integer
+
+-- | a wrapper to signify that the internal PlutusModel
+-- has been updated by the (latest) cardano-protocol changes
+data EvaluationContext = EvaluationContext
+    { toLedgerCostModel   :: LedgerCostModel -- alternative implementation: `extractCostModelParams _pcm`
```

Comment:
The ledger team wants a way to take out the raw Map from the cache, so they can serialize the Map and send it elsewhere. There are 2 possible ways to return them the Map:

1) Cache what they gave to us when they called `mkEvaluationContext`
2) Cache our Plutus.CostModel and compute&return `extractCostModelParams(cached_plutus_costmodel)` 

I opted for (1), but i also cache (2), because it does not that big?

Reply:
They don't need that if they do the "virtual field" approach, because they already have the input to this function stored separately, so why would they need to recompute it? I think we should keep it simple.